[llava][5/N] Add Llava model definition

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4295
* __->__ #4259
* #4256

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

larryliu0820 committed:
[llava][5/N] Add Llava model definition

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

helunwencser commented on examples/models/llava/install_requirements.sh:
```diff
@@ -17,8 +17,14 @@ pip install protobuf
 # Reinstall bitsandbytes to make it compatible.
 pip install bitsandbytes -I
 
+# numpy needs to be pin to 1.24. 1.26.4 will error out
```
missing period.

helunwencser commented on examples/models/llava/install_requirements.sh:
```diff
@@ -17,8 +17,14 @@ pip install protobuf
+# numpy needs to be pin to 1.24. 1.26.4 will error out
+pip install numpy==1.24
+
+# Newer transformer will give TypeError: LlavaLlamaForCausalLM.forward() got an unexpected keyword argument 'cache_position'
```
missing period.

helunwencser commented on examples/models/llava/model.py:
```diff
@@ -4,49 +4,373 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+# An ExecuTorch friendly implementation of Llava-1.5.abs
```
missing period.

helunwencser commented on examples/models/llava/main.py:
```diff
@@ -0,0 +1,56 @@
+    llava = llava_model.get_eager_model()
+
+    prompt_before_image, resized, prompt_after_image = llava_model.get_example_inputs()
+    print(f"Prompt: {llava_model.prompt}")
```
use logging instead?

helunwencser commented on examples/models/llava/main.py:
```diff
@@ -0,0 +1,56 @@
+    outputs = llava_model.tokenizer.batch_decode(output_ids, skip_special_tokens=True)[
+        0
+    ].strip()
+    print(f"Reference output: {outputs}")
```
same

helunwencser commented on examples/models/llava/model.py:
```diff
@@ -4,49 +4,373 @@
+
+    def _get_image_dynamic_shapes(self):
+        height = Dim("height", min=8, max=336)
+        # height = Dim("height", min=8, max=4091)
```
why keep these commented code here?

helunwencser reviewed (approved):
LGTM. Had some minor comments.

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 committed:
Update on "[llava][5/N] Add Llava model definition"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D59759978](https://our.internmc.facebook.com/intern/diff/D59759978)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59759978).
